### PR TITLE
Update rules for removing table markup. Fix #1710 

### LINKED ISSUE #1710: segment_wiki doesn't extract only plain text
The `segment_wiki.py` script doesn't filter formatting properly. Its plain text output contains rubbish like `{| class=\"wikitable\" style=\"text-align: center`, for example in the article `Academy Award for Best Production Design`, but also others.

This is contrary to the stated mission of the script.

### PR DESCRIPTION
**Pull request by chaitaliSaini**
Updated the regex rules for removing table markup.

### CODE REVIEW

File: gensim/corpora/wikicorpus.py [hunk 1]
```diff
@@ -185,10 +186,14 @@ def remove_markup(text, promote_remaining=True, simplify_links=True):
         if simplify_links:
             text = re.sub(RE_P6, '\\2', text)  # simplify links, keep description only
         # remove table markup
-
-        text = text.replace('||', '\n|')  # each table cell on a separate line
+        text = text.replace("!!", "\n|")  # leave only cell content(in table head)
```
Comment (menshikh-iv) [inline gensim/corpora/wikicorpus.py]: What's a performance difference after this changes?

File: gensim/test/test_corpora.py [hunk 2]
```diff
@@ -645,6 +645,18 @@ def test_max_token_len_set(self):
         corpus = self.corpus_class(self.enwiki, processes=1, token_max_len=16, lemmatize=False)
         self.assertTrue(u'collectivization' in next(corpus.get_texts()))
 
+    def test_removed_table_markup(self):
```
Comment (menshikh-iv) [inline gensim/test/test_corpora.py]: Looks like CI stuck on this test (and this failed), please debug it.

**Comment by menshikh-iv** [comment]
ping @chaitaliSaini, how is going?

**Comment by chaitaliSaini** [comment]
Time taken before changes: 2hrs, after changes: 2hrs 22mins.

File: gensim/corpora/wikicorpus.py [hunk 3]
```diff
@@ -68,17 +68,18 @@
...
+RE_P13 = re.compile(r'(?<=(\n[ ])|(\n\n)|([ ]{2})|(.\n)|(.\t))(\||\!)([^[\]\n]*?\|)*', re.UNICODE)
 """Table cell formatting."""
 RE_P14 = re.compile(r'\[\[Category:[^][]*\]\]', re.UNICODE)
 """Categories."""
 RE_P15 = re.compile(r'\[\[([fF]ile:|[iI]mage)[^]]*(\]\])', re.UNICODE)
 """Remove File and Image templates."""
 RE_P16 = re.compile(r'\[{2}(.*?)\]{2}', re.UNICODE)
 """Capture interlinks text and article linked"""
-
+RE_P17 = re.compile(r'(\n.{0,4}((bgcolor)|(\d{0,1}[ ]?colspan)|(rowspan)|(style=)|(class=)|(align=)|(scope=))(.*))|(^.{0,2}((bgcolor)|(\d{0,1}[ ]?colspan)|(rowspan)|(style=)|(class=)|(align=))(.*))', re.UNICODE)
```
Comment (menshikh-iv) [inline gensim/corpora/wikicorpus.py]: PEP8: too long line https://travis-ci.org/RaRe-Technologies/gensim/jobs/351361148#L510, please fix this (we using 120 characters limit)

**Comment by menshikh-iv** [comment]
@piskvorky this change ~20% slowdown `segment_wiki` (but fix bug with tables markup), I think we should merge this (because removing table markup is important, despite the loss in performance), wdyt?

**Comment by piskvorky** [comment]
Of course. Correctness comes before speed.

**Comment by menshikh-iv** [comment]
@piskvorky :+1:
@chaitaliSaini I need some time for check diff in output files before/after (I want to be sure that all works as expected), sorry for waiting

**Comment by menshikh-iv** [comment]
I compared results before and after this change
- this remove markup 
- sometimes this produce too much `\n` (but I don't think that this is a problem).

LGTM, good work @chaitaliSaini :+1:

**Pull request merged**